byterun/weak.c

```diff
@@ -198,167 +243,215 @@ CAMLprim value caml_weak_set (value ar, value n, value el){
   return caml_ephe_set_key_option(ar,n,el);
 }
 
-CAMLprim value caml_ephe_set_data (value ar, value el)
+CAMLexport void caml_ephemeron_set_data (value ar, value el)
 {
-                                                   Assert (Is_in_heap (ar));
+  Assert_valid_ephemeron(ar);
+
   if (caml_gc_phase == Phase_clean){
     /* During this phase since we don't know which ephemeron have been
        cleaned we always need to check it. */
     caml_ephe_clean(ar);
   };
-  do_set (ar, 1, el);
-  return Val_unit;
+  do_set (ar, CAML_EPHE_DATA_OFFSET, el);
 }
 
-CAMLprim value caml_ephe_unset_data (value ar)
+CAMLprim value caml_ephe_set_data (value ar, value el)
 {
-                                                   Assert (Is_in_heap (ar));
-  Field (ar, CAML_EPHE_DATA_OFFSET) = caml_ephe_none;
+  caml_ephemeron_set_data (ar, el);
   return Val_unit;
 }
 
+CAMLexport void caml_ephemeron_unset_data (value ar)
+{
+  Assert_valid_ephemeron(ar);
 
-#define Setup_for_gc
-#define Restore_after_gc
+  Field (ar, CAML_EPHE_DATA_OFFSET) = caml_ephe_none;
+}
 
-CAMLprim value caml_ephe_get_key (value ar, value n)
+CAMLprim value caml_ephe_unset_data (value ar)
 {
-  CAMLparam2 (ar, n);
-  mlsize_t offset = Long_val (n) + 2;
-  CAMLlocal2 (res, elt);
-                                                   Assert (Is_in_heap (ar));
-  if (offset < 2 || offset >= Wosize_val (ar)){
-    caml_invalid_argument ("Weak.get_key");
+  caml_ephemeron_unset_data (ar);
+  return Val_unit;
+}
+
+static value optionalize(int status, value *x) {
```

Comment:
There are a lot of occurrences of "{" on the same line as the declaration of a function, which is a non-standard C coding style, and doesn't match the other functions around.  Can I suggest you put these on the following line?